IdentityCore/src/cache/ios/MSIDKeychainTokenCache.m

```diff
@@ -382,6 +385,51 @@ - (BOOL)saveAppMetadata:(MSIDAppMetadataCacheItem *)item
     return appMetadataitems;
 }
 
+
+- (BOOL)saveAccountMetadata:(MSIDAccountMetadataCacheItem *)item
+                        key:(MSIDAccountMetadataCacheKey *)key
+                 serializer:(id<MSIDAccountMetadataCacheItemSerializer>)serializer
+                    context:(id<MSIDRequestContext>)context
+                      error:(NSError **)error
+{
+
+    if (!item)
+    {
+        if (error)
+        {
+            NSString *errorMessage = @"Nil metadata item is received!";
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, errorMessage, nil, nil, nil, context.correlationId, nil);
```

Comment:
I think this error will map to MSALErrorInternal error with MSALInternalErrorCodeKey == MSALErrorInternal. Seems I missed this case when I split errors. I think we should either add a new key to MSALInternalError, which we will map to MSIDErrorInternal or avoid using MSIDErrorInternal and specify a concrete error instead (btw, we already using MSIDErrorInternal in few other places, so we will need to update them too).

Reply:
@antrix1989 can you create an issue so we keep track? Thanks